nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,160 @@
+        '';
+      };
+      port = mkOption {
+        type = types.port;
+        default = 9000;
+        description = mdDoc "The port webhook should be reachable from.";
+      };
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Open the configured port in the firewall for external ingress traffic.
+          Preferably the Webhook server is instead put behind a reverse proxy.
+        '';
+      };
+      enableTemplate = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc "Enable the generated hooks file to be parsed as a Go template.";
+      };
+      urlPrefix = mkOption {
+        type = types.str;
+        default = "hooks";
+        description = mdDoc ''
+          The URL path prefix to use for served hooks
+
+          Example: protocol://yourserver:port/PREFIX/:hook-id
+        '';
+      };
+      hooks = mkOption {
+        type = types.str;
+        default = ''
+        [{
+          "id": "echo",
+          "execute-command": "echo",
+          "response-message": "Webhook is reachable!"
+        }]'';
+        description = mdDoc ''
+          The actual configuration of which hooks will be served.
+
+          It cannot be empty otherwise your service won't start.
+          Read more on the [project homepage] and on the [hook definition] page.
+
+          [hook definition]: https://github.com/adnanh/webhook/blob/master/docs/Hook-Definition.md
+          [project homepage]: https://github.com/adnanh/webhook#configuration
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users = mkIf (cfg.user == defaultUser) {
+      ${defaultUser} =
+        {
+          isSystemUser = true;
+          group = cfg.group;
+          description = "Webhook daemon user";
+        };
+    };
+
+    users.groups = mkIf (cfg.user == defaultUser && cfg.group == defaultUser) {
+      ${defaultUser} = {};
+    };
+
+    networking.firewall = lib.mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.port ];
+    };
+
+    systemd.services = {
+      webhook = {
+        description = "Webhook service";
+        after = [ "network.target" ];
+        environment = {
+          STNORESTART = "yes";
+          STNOUPGRADE = "yes";
+          inherit (cfg);
+        } // config.networking.proxy.envVars;
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          Restart = "on-failure";
+          SuccessExitStatus = "2 3 4";
+          RestartForceExitStatus="3 4";
+          User = cfg.user;
+          Group = cfg.group;
+          ExecStart = ''
+            ${cfg.package}/bin/webhook \
+              -hooks ${hooksFile} \
+              ${if cfg.enableTemplate then "-template" else ""} \
+              -ip ${cfg.ip} \
+              -port ${toString cfg.port} \
+              -urlprefix ${cfg.urlPrefix}
+          '';
```

Comment:
We might want to add an `extraArgs` option for arguments that are not supported by the module.

Reply:
Done in  1fa6933